### PR TITLE
FR: Add Stats by Current Roster #239

### LINKED ISSUE #239: Possible to get Stats/Matches/Etc for a teams current lineup only?
Hey,

Is it possible to get the stats and match history and what not for a teams current lineup? Kinda how now on hltv, you can go to a team's page, and where it shows their current lineup, click on 'Replace context with lineup' and then browse their history, stats, etc based just on the current 5 man roster. 

Thanks,
David

### PR DESCRIPTION
**Pull request by dbalders**
* Update getTeamStats to allow stats for the current roster
    * Add new boolean param
    * Add url variables to allow changing between current lineup and all time
    * Add logic for if currentRoster is true

Closes #239

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -270,6 +270,7 @@ Parses the info from the `hltv.org/stats/teams/*` page (4 requests)
 | Option |  Type  | Default value | Description |
 | :----: | :----: | :-----------: | :---------: |
 |   id   | number |       -       | The team id |
+|   currentRoster   | boolean |       false       | Only Current Roster |
```
Comment (gigobyte) [inline README.md:273]: Can you change it to `currentRosterOnly`? I feel like it's a bit more descriptive.

File: src/endpoints/getTeamStats.ts [hunk 2]
```diff
@@ -85,7 +116,7 @@ export const getTeamStats = (config: HLTVConfig) => async ({
     place: eventEl.find('.statsCenterText').text(),
     event: {
       id: Number(
-        eventEl.find('.image-and-label').first().attr('href')!.split('=')[1]
+        eventEl.find('.image-and-label').first().attr('href')!.split('=')[eventEl.find('.image-and-label').first().attr('href')!.split('=').length - 1]
```
Comment (gigobyte) [inline src/endpoints/getTeamStats.ts:119]: What is the reason for this change?

File: src/endpoints/getTeamStats.ts [hunk 3]
```diff
@@ -4,24 +4,75 @@ import { fetchPage, toArray, getTimestamp, getMapSlug } from '../utils/mappers'
...
+  )
+  const standins = getPlayersByContainer(getContainerByText('Standins'))
+
+  if (currentRoster) {
+    currentRosterURL = 'lineup=' + currentLineup[0].id
+    currentRosterURL = currentRosterURL + '&lineup=' + currentLineup[1].id
+    currentRosterURL = currentRosterURL + '&lineup=' + currentLineup[2].id
+    currentRosterURL = currentRosterURL + '&lineup=' + currentLineup[3].id
+    currentRosterURL = currentRosterURL + '&lineup=' + currentLineup[4].id
+    currentRosterURL = currentRosterURL + '&minLineupMatch=0'
```
Comment (gigobyte) [inline src/endpoints/getTeamStats.ts:50]: I feel like this can be refactored. Something like:
```js
currentRosterUrl = `lineup=${currentLineup.map(x => x.id).join('&lineup=')}&minLineupMatch=0`
```

File: src/endpoints/getTeamStats.ts (hunk 2 above)
Comment (dbalders) [reply src/endpoints/getTeamStats.ts:119]: For this, when doing the current roster, the `split('=')[1]` is wrong because the url contains 5 = signs because of the `?lineup=` in the url. So, it wasn't finding the event id. This change makes it so it just grabs the id after the last = sign. Does that explanation make sense?

File: README.md (hunk 1 above)
Comment (dbalders) [reply README.md:273]: Yep!

File: src/endpoints/getTeamStats.ts (hunk 3 above)
Comment (dbalders) [reply src/endpoints/getTeamStats.ts:50]: smart

**Comment by dbalders** [comment]
Updated

**Review by gigobyte — APPROVED** [review]
